src/components/App.jsx

```diff
@@ -141,43 +179,43 @@ function App() {
                 </> 
                } 
                />
-                  <Route 
-                    path="/saved-articles" 
-                    element={
-                      <>
-                      <Header 
-                      handleOpenLoginPopup={handleOpenLoginPopup} 
+                  <Route element={<ProtectedRoute
                       isLoggedIn={isLoggedIn} 
-                      currentUser={currentUser} 
-                      handleLogout={handleLogout} 
+                      handleOpenLoginPopup={handleOpenLoginPopup}/>}>
+                    <Route 
+                      path="/saved-articles" 
+                      element={
+                        <>
+                        <Header 
+                        handleOpenLoginPopup={handleOpenLoginPopup} 
+                        isLoggedIn={isLoggedIn} 
+                        currentUser={currentUser} 
```

Comment:
`currentUser={currentUser}` is not needed in the props because you should take it from `currentUserContext`